### PR TITLE
Fix link icons

### PR DESCRIPTION
**Pull request by kurtw**
The logic handling links doesn't insert the correct markup.

## Additions

- Added download link regex pattern and logic to parse_link function for rich text.

## Removals

- CSS to add icon classes for files

## Changes

- Markup is inserted and manipulated differently for affected links matching the patterns.

## Testing

- Add cfpb link
- Add .gov link
- Add external link
- add file link
- make sure markup is correct for each one

## Review

- @Scotchester 
- @willbarton 
- @richaagarwal 
- @virginiacc 
- @contolini 

## Notes

-

## Todos

- Change where logic is located for efficiency
- Possible refactor for readability
- Write tests

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by Scotchester** [comment]
@kurtw The Travis failure indicates a problem when trying to run `parse_links` within the `wordpress_sub_page.py` processor.

Relevant bits of the log:

```bash
  File "/home/travis/build/cfpb/cfgov-refresh/cfgov/processors/wordpress_sub_page.py", line 26, in documents
    yield process_sub_page(post)
  File "/home/travis/build/cfpb/cfgov-refresh/cfgov/processors/wordpress_sub_page.py", line 71, in process_sub_page
    post = process_external_links(post)
  File "/home/travis/build/cfpb/cfgov-refresh/cfgov/sheerlike/external_links.py", line 9, in process_external_links
    doc[key] = _process_data(value)
  File "/home/travis/build/cfpb/cfgov-refresh/cfgov/sheerlike/external_links.py", line 17, in _process_data
    field = parse_links(soup).encode(formatter=None)
  File "/home/travis/build/cfpb/cfgov-refresh/cfgov/v1/__init__.py", line 93, in parse_links
    a.attrs['class'] = ' '.join([external_a_class, a.attrs['class']])
TypeError: sequence item 1: expected string, list found
```

**Comment by kurtw** [comment]
Try it now @Scotchester

**Comment by Scotchester** [comment]
Travis likes it! I'll pull it down and check locally.

**Comment by Scotchester** [comment]
Pushed a commit to fix a few issues. See commit message for details.

Also, I thought of a new todo that's out-of-scope for this PR: We should be applying this to URLs that are entered into the URL field for links in the CMS. For example, the link in a Hero, FCM, or a list of Related Links. Since those aren't in RichText fields, they aren't affected by this code.

**Comment by kurtw** [comment]
Good addition @Scotchester. I made a slight change to clean it a tad. Looks good to me 👍

**Comment by Scotchester** [comment]
LOL, obvious improvement. Nice. :+1:

Anyone else want to test?

**Comment by rosskarchner** [comment]
Looks good here: 👍

File: cfgov/v1/__init__.py [hunk 1]
```diff
@@ -77,16 +79,35 @@ def parse_links(soup):
...
-            a.append(' ')  # We want an extra space before the icon
-            a.append(soup.new_tag('span', attrs='class="%s"' % span_class))
-        elif not files_pattern.match(a['href']):
-            fix_link(a)
+        needs_icon = True
+        for child in a.children:
+            if child.name in ['img', 'svg']:
+                needs_icon = False
+                break
+        if needs_icon:
```
Comment (vccabral) [inline cfgov/v1/__init__.py]: Break this method down.

**Comment by Scotchester** [comment]
@kurtw That Travis error looks legit.

**Comment by kurtw** [comment]
Odd. I thought I fixed that before pushing.

**Comment by Scotchester** [comment]
Interesting. I didn't get an email notification about that new commit. Possibly because you amended it?

Anyway, Travis likes it now. I will test locally once more.

**Comment by Scotchester** [comment]
Still working well! :+1:

**Comment by Scotchester** [comment]
One quick note, though, just saw this in my console output from when I tested that. Not sure it's a big deal, but take a look, just in case:

```
/Users/cranfills/.virtualenvs/cfgov-refresh/lib/python2.7/site-packages/bs4/__init__.py:166: UserWarning: No parser was explicitly specified, so I'm using the best available HTML parser for this system ("lxml"). This usually isn't a problem, but if you run this code on another system, or in a different virtual environment, it may use a different parser and behave differently.

To get rid of this warning, change this:

 BeautifulSoup([your markup])

to this:

 BeautifulSoup([your markup], "lxml")

  markup_type=markup_type))
```

**Comment by chosak** [comment]
@kurtw I get three test failures because of these changes when I try to run the code in both `master` and in PR #2244. It appears that commit 48ea4b6 broke these tests (see the Travis log [here](https://travis-ci.org/cfpb/cfgov-refresh/builds/147553132#L5191) but for some reason Travis still marked the build as passing.

The test failures are all in `v1.tests.test_links.ImportDataTest` (run quickly with `tox -e fast v1.tests.test_links`) and at first glance appear to be because the `parse_links` function used to take a `BeautifulSoup` object (in the tests written by @richaagarwal) but now seem to take raw text.

**Pull request merged**